### DISCUSSION TITLE
Readable names in text dumps: Better identifiers?

### DISCUSSION THREAD

**Discussion by mootari** [rusq/slackdump#130] [Ideas]
***Edit: Implemented in #132***

I just gave the text output format a try and noticed that the dumps display the <del>first part of a user's email address as a readable identifier. Unfortunately that fails to work if users have addresses at @ own domains, e.g. "info@example.com" or "contact@example.com".</del> <ins>deprecated user name</ins>. Example:
```
> contact [THEUSERID] @ 08/05/2022 12:33:22 Z:
Yeah, I've seen his intro
Doesn't look like he's active though
```

I have no immediate use for the text format, so I've chosen to create this discussion instead of an issue. <del>My recommendation would be to either display the email address in full, or to partially redact it (as is done in Monorail, Chromium's bug tracker):</del>
```
cont...@example.com
```
<del>For reference, this is the monorail implementation: https://chromium.googlesource.com/infra/infra/+/0af18ef4/appengine/monorail/framework/framework_bizobj.py#175</del>

<ins>Edit: These were actually user names, not emails. The current recommendation is to show display names instead.</ins>

**Comment by rusq** [comment]
Hey @mootari, thanks for the feedback!

It uses  the username (user.Name), not email.  To be more precise, to determine the "Sender" name it uses this logic: https://github.com/rusq/slackdump/blob/master/internal/structures/user_index.go#L40

I don't think I ever used user emails anywhere in formatted output.

**Reply by mootari** [reply]
Interesting! User names are non-configurable though, and apparently have been deprecated for quite a while: https://api.slack.com/changelog/2017-09-the-one-about-usernames

So display names might be the better choice?

**Reply by mootari** [reply]
In `slack-go/slack` the display name is part of [UserProfile](https://pkg.go.dev/github.com/slack-go/slack#UserProfile).

*Edit: Luckily the profile is already present in `users.json`, so additional API calls shouldn't be necessary.*

**Reply by rusq** [reply]
Hahaha, deprecated since 2017 and still in use! Yeah, probably better switch to display names, to avoid unfortunate surprises in future

**Reply by rusq** [reply]
Ran some experiments with display and user names, and as a result switched to using RealNames.  DisplayName can be empty, and that produces not very pleasant output

https://github.com/rusq/slackdump/pull/132

**Reply by mootari** [reply]
What about defaulting to DisplayName, but falling back to RealName if it's empty? I think that's also how Slack does it?

**Reply by rusq** [reply]
Good idea, done!
